src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -106,188 +107,195 @@ void EnCow_Init(Actor* thisx, PlayState* play) {
             Animation_PlayLoop(&this->skelAnime, &gCowTailIdleAnim);
-            this->actor.update = func_809DFE98;
-            this->actor.draw = func_809E0070;
-            this->actionFunc = func_809DFA84;
-            func_809DEF94(this);
+            this->actor.update = EnCow_UpdateTail;
+            this->actor.draw = EnCow_DrawTail;
+            this->actionFunc = EnCow_IdleTail;
+            EnCow_SetTailPos(this);
             this->actor.flags &= ~ACTOR_FLAG_0;
-            this->unk_278 = ((u32)(Rand_ZeroFloat(1000.0f)) & 0xFFFF) + 40.0f;
+            this->animationTimer = ((u32)(Rand_ZeroFloat(1000.0f)) & 0xFFFF) + 40.0f;
             break;
     }
     this->actor.colChkInfo.mass = MASS_IMMOVABLE;
     Actor_SetScale(&this->actor, 0.01f);
-    this->unk_276 = 0;
+    this->cowFlags = 0;
 }
 
 void EnCow_Destroy(Actor* thisx, PlayState* play) {
     EnCow* this = (EnCow*)thisx;
 
-    if (this->actor.params == 0) {
-        Collider_DestroyCylinder(play, &this->colliders[0]);
-        Collider_DestroyCylinder(play, &this->colliders[1]);
+    if (EN_COW_TYPE(this) == EN_COW_TYPE_DEFAULT) {
+        Collider_DestroyCylinder(play, &this->colliders[EN_COW_COLLIDER_FRONT]);
+        Collider_DestroyCylinder(play, &this->colliders[EN_COW_COLLIDER_REAR]);
     }
 }
 
-void func_809DF494(EnCow* this, PlayState* play) {
-    if (this->unk_278 > 0) {
-        this->unk_278 -= 1;
+void EnCow_UpdateAnimation(EnCow* this, PlayState* play) {
+    if (this->animationTimer > 0) {
+        this->animationTimer--;
     } else {
-        this->unk_278 = Rand_ZeroFloat(500.0f) + 40.0f;
+        this->animationTimer = Rand_ZeroFloat(500.0f) + 40.0f;
         Animation_Change(&this->skelAnime, &gCowBodyChewAnim, 1.0f, this->skelAnime.curFrame,
                          Animation_GetLastFrame(&gCowBodyChewAnim), ANIMMODE_ONCE, 1.0f);
     }
 
-    if ((this->actor.xzDistToPlayer < 150.0f) && !(this->unk_276 & 2)) {
-        this->unk_276 |= 2;
-        if (this->skelAnime.animation == &gCowBodyChewAnim) {
-            this->unk_278 = 0;
+    if (this->actor.xzDistToPlayer < 150.0f) {
+        if (!(this->cowFlags & EN_COW_FLAG_PLAYER_HAS_APPROACHED)) {
+            this->cowFlags |= EN_COW_FLAG_PLAYER_HAS_APPROACHED;
+            if (this->skelAnime.animation == &gCowBodyChewAnim) {
+                this->animationTimer = 0;
+            }
         }
     }
 
-    this->unk_27A += 1;
-    if (this->unk_27A >= 0x31) {
-        this->unk_27A = 0;
+    this->animationCycle++;
+    if (this->animationCycle > 48) {
+        this->animationCycle = 0;
     }
 
     // (1.0f / 100.0f) instead of 0.01f below is necessary so 0.01f doesn't get reused mistakenly
-    if (this->unk_27A < 0x20) {
-        this->actor.scale.x = ((Math_SinS(this->unk_27A << 0xA) * (1.0f / 100.0f)) + 1.0f) * 0.01f;
+    if (this->animationCycle < 32) {
+        this->actor.scale.x = ((Math_SinS(this->animationCycle * 1024) * (1.0f / 100.0f)) + 1.0f) * 0.01f;
```

Comment:
Didn't you have `breathPhase` or `breathTimer` instead of MM's `animationCycle` at one point? I think it'd be a better name